miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -178,6 +180,69 @@ export.set_account_item
     # => [R', V]
 end
 
+#! Returns VALUE located under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds.
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, KEY, ...]
+#! Output: [VALUE]
+#!
+#! - index is the index of the item to get.
+#! - VALUE is the value of the item.
+export.get_account_map_item
+    # check if storage type is map
+    dup exec.account::get_storage_slot_type_info drop
+    # => [slot_type, index, KEY, ...]
+
+    # fails if slot_type is not 1 = map
+    assert
+    # => [index, KEY, ...]
```

Comment:
maybe lets add an explicit equality check? E.g.:

```
push.MAP_TYPE eq assert
```

That way we can change the constant for the `MAP_TYPE` down the line, and this will be automatically updated (otherwise it is easy to miss this and it becomes a bug).